src/App.jsx

```diff
@@ -122,43 +127,47 @@ function App() {
-                  <ProtectedRouteWithProps path="/sites/:siteName/pages/:fileName" component={EditPage} isCollectionPage={false} isResourcePage={false} />
-                  <ProtectedRouteWithProps path="/sites/:siteName/workspace" component={Workspace} />
-                  <ProtectedRouteWithProps path="/sites/:siteName/homepage" component={EditHomepage} />
-                  <ProtectedRouteWithProps path="/sites/:siteName/contact-us" component={EditContactUs} />
-                  <ProtectedRouteWithProps path="/sites/:siteName/resources/:resourceName/:fileName" component={EditPage} isCollectionPage={false} isResourcePage={true} />
-                  <ProtectedRouteWithProps path="/sites/:siteName/resources/:collectionName" component={CategoryPages} isResource={true}/>
-                  <ProtectedRouteWithProps path="/sites/:siteName/resources" component={Resources} />
-                  <ProtectedRouteWithProps path="/sites/:siteName/menus/main-menu" component={EditNav} />
-                  <ProtectedRouteWithProps path="/sites/:siteName/menus" component={Menus} />
-                  <ProtectedRouteWithProps path="/sites/:siteName/settings" component={Settings} />
-                  <ProtectedRouteWithProps exact path="/sites" component={Sites} />
-                  <ProtectedRouteWithProps path="/" component={NotFoundPage}/>
-                  <Route>
-                    <Redirect to={ isLoggedIn ? '/sites' : '/' } />
-                  </Route>
-              </Switch>
-            </LoginContext.Provider>
-        </div>
+      <QueryClientProvider client={queryClient}>
+          <ToastContainer hideProgressBar={true} position='top-center' closeButton={ToastCloseButton} className={elementStyles.toastContainer}/>
+          <div>
+            {/*
+              A <Switch> looks through all its children <Route>
+              elements and renders the first one whose path
+              matches the current URL. Use a <Switch> any time
+              you have multiple routes, but you want only one
+              of them to render at a time
+            */}
+              <LoginContext.Provider value={{isLoggedIn, setLogin, setLogoutState}}>
+                <Switch>
+                    <ProtectedRouteWithProps exact path='/auth' component={AuthCallback} />
+                    <ProtectedRouteWithProps exact path="/" component={Home} />
+                    <ProtectedRouteWithProps path="/sites/:siteName/collections/:collectionName/:fileName" component={EditPage} isCollectionPage={true} isResourcePage={false} />
+                    <ProtectedRouteWithProps path="/sites/:siteName/collections/:collectionName" component={CategoryPages} isResource={false}/>
+                    <ProtectedRouteWithProps exact path="/sites/:siteName/folder/:folderName" component={Folders} />
+                    <ProtectedRouteWithProps exact path="/sites/:siteName/folder/:folderName/subfolder/:subfolderName" component={Folders} />
```

Comment:
will this artificially restrict us to 1 level of nesting?

Reply:
As discussed, since we do not foresee the Isomer template accommodating more than a third-nav level of nesting, we'll stick with the current routing